lib/base/index.js

```diff
@@ -1,136 +1,138 @@
+    if (!username || typeof username !== 'string') {
       throw new Error(`Please provide a username to initialize the bot. Got: ${JSON.stringify(username)}`)
-    // Don't want to accidentally print the paperkey to STDERR
-    if (!paperkey || typeof paperkey !== 'string')
+    }
+    if (!paperkey || typeof paperkey !== 'string') {
+      // Don't want to accidentally print the paperkey to STDERR.
       throw new Error(`Please provide a paperkey to initialize the bot.`)
-
-    const update = {
-      homeDir: randomTempDir(),
     }
-    store.dispatch(actionPreinit(update))
+    const homeDir = (options && options.homeDir) || randomTempDir()
+    this.homeDir = homeDir
 
-    const {homeDir} = store.getState()
     await keybaseServiceStartup(homeDir)
-    await keybaseExec(homeDir, ['oneshot', '--username', username], {
-      stdinBuffer: paperkey,
-    })
+    await keybaseExec(homeDir, ['oneshot', '--username', username], {stdinBuffer: paperkey})
 
-    const currentDPair = await getCurrentUsernameAndDevicename(store)
+    const currentDPair = await this._currentUsernameAndDevice()
     if (currentDPair && currentDPair.username && currentDPair.devicename) {
-      const update = {
-        username: currentDPair.username,
-        devicename: currentDPair.devicename,
-        flags: {
-          verbose: options ? Boolean(options.verbose) : false,
-        },
-      }
-      store.dispatch(actionInit(update))
-      // TODO: @jacob - refactor logging to depend on verbose flag in state
+      this.initialized = 'paperkey'
+      this.username = currentDPair.username
+      this.devicename = currentDPair.devicename
+      this.verbose = options ? Boolean(options.verbose) : false
+      // TODO: @jacob - refactor logging
       // this._log(`intialized ${currentDPair.username} (device=${currentDPair.devicename})`)
     }
   }
-  return init
-}
 
-const _deinit = (store: Store) => {
   /**
+   * Initialize a Keybase bot off of a currently running service.
+   * @memberof BaseBot
+   */
+  async initWithExistingService(options?: InitOptions) {
+    if (options && options.homeDir) {
+      this.homeDir = options.homeDir
+    }
+    const currentDPair = await this._currentUsernameAndDevice()
+    if (currentDPair && currentDPair.username && currentDPair.devicename) {
+      this.initialized = 'currentService'
+      this.username = currentDPair.username
+      this.devicename = currentDPair.devicename
+      this.verbose = options ? Boolean(options.verbose) : false
+      // TODO: @jacob - refactor logging
+      // this._log(`intialized ${currentDPair.username} (device=${currentDPair.devicename})`)
+    }
+  }
+
+  /**
+   * Deinitializes the bot by logging out, stopping the keybase service, and removing any leftover login files made by the bot. This should be run before your bot ends.
+   * @memberof BaseBot
+   */
+  async deinit(): Promise<void> {
+    if (this.initialized === 'paperkey') {
```

Comment:
It seems fair to only logout and stop the service if a user `init`ed the standard way. If they use an existing service, I think we want to do nothing on a `deinit`

Reply:
agreed, with one exception. we still need to kill any processes that were spawned by `watchFor...` calls.